src/ui/public/agg_types/buckets/terms_other_bucket_helper.js

```diff
@@ -0,0 +1,143 @@
+import _ from 'lodash';
+import { VisAggConfigProvider } from 'ui/vis/agg_config';
+import { migrateFilter } from 'ui/courier/data_source/_migrate_filter';
+import { buildPhrasesFilter } from 'ui/filter_manager/lib/phrases';
+import { buildExistsFilter } from 'ui/filter_manager/lib/exists';
+
+const getAggConfig = (aggs, aggWithOtherBucket) => {
+  if (aggs[aggWithOtherBucket.id]) return aggs[aggWithOtherBucket.id];
+  return getAggConfig(_.values(aggs)[0].aggs, aggWithOtherBucket);
+};
+
+const getAggResultBuckets = (aggsConfig, response, aggWithOtherBucket, key) => {
+  const keyParts = key.split('-');
+  let responseAgg = response;
+  for (const i in keyParts) {
+    if (keyParts[i]) {
+      const agg = _.values(responseAgg)[0];
+      const aggKey = _.keys(responseAgg)[0];
+      const aggConfig = _.find(aggsConfig, agg => agg.id === aggKey);
+      const bucket = _.find(agg.buckets, (bucket, bucketObjKey) => {
+        const bucketKey = aggConfig.getKey(bucket, Number.isInteger(bucketObjKey) ? null : bucketObjKey).toString();
+        return bucketKey === keyParts[i];
+      });
+      if (bucket) {
+        responseAgg = bucket;
+      }
+    }
+  }
+  if (responseAgg[aggWithOtherBucket.id]) return responseAgg[aggWithOtherBucket.id].buckets;
+  return [];
+};
+
+const getAggConfigResult = (responseAggs, aggId, bucketKey) => {
+  let resultBuckets = [];
+  if (responseAggs[aggId]) {
+    const matchingBucket = responseAggs[aggId].buckets.find(bucket => bucket.key === bucketKey);
+    if (matchingBucket) resultBuckets.push(matchingBucket);
+    return resultBuckets;
+  }
+  _.each(responseAggs, agg => {
+    resultBuckets = [
+      ...resultBuckets,
+      ...getAggConfigResult(agg.aggs, aggId, bucketKey)
```

Comment:
we probably don't want to recurse if we don't have agg.aggs. Or put a guard on top. in any case, we probably want a base-case in this recursive function somewhere.